core/src/modules/ops/destination.ts

```diff
@@ -350,6 +350,16 @@ export namespace DestinationOps {
           )[0].remoteKey
       );
 
+    const syncMode = await destination.getSyncMode();
+    const canDelete = syncMode
+      ? DestinationSyncModeData[syncMode].operations.delete
```

Comment:
Confirming that `DestinationSyncModeData[syncMode].operations.delete` can never be null.  If so, this wold be true if undefined

Reply:
Hmm, not sure I understand what you mean. What would be `true` if what is `undefined`?

Reply:
sorry about that! I read this logic to say that `canDelete` would be `true` if `DestinationSyncModeData[syncMode].operations === null || undefined`.  I don't think that's possible, but i wanted to confirm.

Reply:
`undefined` isn't possible because of the type on DestinationSyncModeData, but `null` is technically valid. If the operations are `null` then this would error here saying you can't access `delete` of `null`. We're the ones defining what each syncMode does in core though so this shouldn't be possible.